tiatoolbox/visualization/tileserver.py

```diff
@@ -2,15 +2,17 @@
 import io
 import json
 from pathlib import Path
-from typing import Dict
+from typing import Dict, List, Union
 
 import numpy as np
 from flask import Flask, Response, send_file
 from flask.templating import render_template
+from PIL import Image
 
 from tiatoolbox import data
 from tiatoolbox.tools.pyramid import ZoomifyGenerator
-from tiatoolbox.wsicore.wsireader import WSIReader
+from tiatoolbox.wsicore.wsireader import VirtualWSIReader, WSIReader
+from tiatoolbox.utils.visualization import colourise_image
 
 
 class TileServer(Flask):
```

Comment:
Might need to add to the docstring here to explain new behaviour, e.g. how layer names are determined and coloursiation etc.